metals/src/main/scala/scala/meta/internal/metals/Time.scala

```diff
@@ -1,17 +1,17 @@
 package scala.meta.internal.metals
 
-import java.util.concurrent.TimeUnit
-
 /**
- * Wrapper around `System.nanoTime` to allow easier testing.
+ * Wrapper around `System.nanoTime` and `System.currentTimeMillis`
+ * to allow easier testing.
  */
 trait Time {
   def nanos(): Long
-  final def millis(): Long = TimeUnit.NANOSECONDS.toMillis(nanos())
+  def millis(): Long
 }
 
 object Time {
   object system extends Time {
     def nanos(): Long = System.nanoTime()
```

Comment:
We can also include the "absolute time base" for proper calculation of the elapsed nanoseconds:
```suggestion
    private val absoluteTimeBaseNanos: Long = {
      val now = Instant.now()
      val seconds = TimeUnit.SECONDS.toNanos(now.getEpochSecond)
      val nanos = now.getNano
      seconds + nanos
    }

    private val relativeTimeBaseNanos = System.nanoTime()

    def nanos(): Long = {
      absoluteTimeBaseNanos + elapsed
    }

    private def elapsed: Long = System.nanoTime() - relativeTimeBaseNanos
```


Note, that this does not handle the time zones at all. Probably it is safe to assume the time of the server as the valid one. At least for now

Reply:
I think it might be too much complexity for now.